editoast/editoast_derive/src/error.rs

```
@@ -62,6 +68,48 @@ pub fn expand_editoast_error(input: &DeriveInput) -> Result<TokenStream> {
     })
 }
 
+fn parse_error_definition(
+    base_id: String,
+    default_status: u16,
+    variant: &ParsedVariant,
+) -> TokenStream {
+    //Error name
+    let name = variant.ident.unraw().to_string();
+
+    // Compute its id
+    let id = format!("editoast:{}:{}", base_id, name);
+
+    // Retieve error status (or get the default one)
+    let status = match variant.params.status.as_ref() {
+        Some(syn::Expr::Lit(exprlit)) => match &exprlit.lit {
+            Lit::Int(lit) => lit.base10_parse::<u16>().unwrap(),
+            _ => default_status,
+        },
+        _ => default_status,
+    };
+
+    // Retrieve the list of parameters that are given in the error
+    let mut context = std::collections::HashMap::new();
+    variant.fields.iter().for_each(|field| {
+        match field.ident.as_ref() {
+            Some(name) => {
+                context.insert(
+                    name.to_string(),
+                    extract_type(&field.ty).unwrap_or(String::from("String")),
```

Shouldn't we return an error instead of using `String`?

---

done but need to be checked

---

Using expect is not the best thing to do. Ideally, we return a `darling::Result<TokenStream>`. A darling error contains contains a token location so the user can have a precise linter error.

---

Thanks for the tip, will check that

---

Un fix it, can you check it ? Thanks !